docs/versioned_docs/version-v1.3.0/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -0,0 +1,599 @@
+
+        // Add Service Bus Topic message pump and use OrdersMessageHandler to process the messages on the 'My-Subscription-Name' subscription
+        // - Topic subscriptions over 50 characters will be truncated
+        // - ISecretProvider will be used to lookup the connection string scoped to the queue for secret ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING
+        services.AddServiceBusTopicMessagePump("My-Subscription-Name", "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING")
+                .WithServiceBusMessageHandler<OrdersMessageHandler, Order>();
+
+        // Note, that only a single call to the `.WithServiceBusMessageHandler` has to be made when the handler should be used across message pumps.
+    }
+}
+```
+
+In this example, we are using the Azure Service Bus message pump to process a queue and a topic and use the connection string stored in the `ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING` connection string.
+
+> 💡 We support **connection strings that are scoped on the Service Bus namespace and entity** allowing you to choose the required security model for your applications. If you are using namespace-scoped connection strings you'll have to pass your queue/topic name as well.
+
+### Filter messages based on message context
+When registering a new message handler, one can opt-in to add a filter on the message context which filters out messages that are not needed to be processed.
+
+This can be useful when you are sending different message types on the same queue. Another use-case is being able to handle different versions of the same message type which have different contracts because you are migrating your application.
+
+Following example shows how a message handler should only process a certain message when a property's in the context is present.
+
+We'll use a simple message handler implementation:
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions.MessagingHandling;
+
+public class OrderMessageHandler : IAzureServiceBusMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, AzureServiceBusMessageContext context, ...)
+    {
+        // Do some processing...
+    }
+}
+```
+
+We would like that this handler only processed the message when the context contains `MessageType` equals `Order`.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.WithServiceBusMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
+    }
+}
+```
+
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
```

Comment:
I already answered this question in a preview PR 😄 .

The `may` here has to do with how the message handler is registered. Yes, the order of the registration matters. First one registered is the first one that gets checked (this can be clearly seen in the tests). No, it is not random, so yes it in your example, the first handler will always be checked first.

The `may` here is about how it registered. If there would be a message handler above the one in the sample that has more flexible filters (for example), then that one would be chosen.

Reply:
I'll add an additional sentence to stress this deterministic behavior.

Reply:
https://github.com/arcus-azure/arcus.messaging/pull/337/commits/f5763a9949016e4d29c401337af5a26c2f3f0eba

Reply:
Indeed, I remember asking that.  But, I think it is good that it has now been made a bit more clear :)